lxd/instance_efi_vars.go

```diff
@@ -0,0 +1,218 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+
+	"github.com/gorilla/mux"
+
+	"github.com/canonical/lxd/lxd/instance"
+	"github.com/canonical/lxd/lxd/instance/instancetype"
+	"github.com/canonical/lxd/lxd/request"
+	"github.com/canonical/lxd/lxd/response"
+	"github.com/canonical/lxd/lxd/util"
+	"github.com/canonical/lxd/shared"
+	"github.com/canonical/lxd/shared/api"
+)
+
+// swagger:operation GET /1.0/instances/{name}/efi_vars instances instance_efi_vars_get
+//
+//	Get the VM instance EFI variables
+//
+//	Gets a specific instance EFI variables.
+//
+//	---
+//	produces:
+//	  - application/json
+//	parameters:
+//	  - in: query
+//	    name: project
+//	    description: Project name
+//	    type: string
+//	    example: default
+//	responses:
+//	  "200":
+//	    description: Instance EFI variables
+//	    schema:
+//	      type: object
+//	      description: Sync response
+//	      properties:
+//	        type:
+//	          type: string
+//	          description: Response type
+//	          example: sync
+//	        status:
+//	          type: string
+//	          description: Status description
+//	          example: Success
+//	        status_code:
+//	          type: integer
+//	          description: Status code
+//	          example: 200
+//	        metadata:
+//	          $ref: "#/definitions/InstanceEFI"
+//	  "403":
+//	    $ref: "#/responses/Forbidden"
+//	  "500":
+//	    $ref: "#/responses/InternalServerError"
+func instanceEFIVarsGet(d *Daemon, r *http.Request) response.Response {
+	s := d.State()
+
+	instanceType, err := urlInstanceTypeDetect(r)
+	if err != nil {
+		return response.SmartError(err)
+	}
+
+	projectName := request.ProjectParam(r)
+	name, err := url.PathUnescape(mux.Vars(r)["name"])
+	if err != nil {
+		return response.SmartError(err)
+	}
+
+	if shared.IsSnapshot(name) {
+		return response.BadRequest(fmt.Errorf("Invalid instance name"))
+	}
+
+	// Handle requests targeted to a container on a different node
+	resp, err := forwardedResponseIfInstanceIsRemote(s, r, projectName, name, instanceType)
+	if err != nil {
+		return response.SmartError(err)
+	}
+
+	if resp != nil {
+		return resp
+	}
+
+	inst, err := instance.LoadByProjectAndName(s, projectName, name)
+	if err != nil {
+		return response.SmartError(err)
+	}
+
+	if inst.Type() != instancetype.VM {
+		return response.BadRequest(fmt.Errorf("EFI variables manipulation supported for VM type instances only"))
+	}
+
+	instanceEFI, err := inst.EFIVarsRead()
+	if err != nil {
+		return response.SmartError(err)
+	}
+
+	etag := []any{inst.Architecture(), inst.LocalConfig(), instanceEFI}
```

Comment:
Why are inst.Architecture and inst.LocalConfig included for this etag out of interest?

Reply:
Probably I was over cautious :-) We can just use `instanceEFI` to generate ETag, for sure.

Reply:
Have fixed.